rfd/0156-notifications.md

```diff
@@ -103,8 +102,21 @@ deleted as well as its state item.
 // UserNotificationState represents a notification's state for a user. This is to keep track
 // of whether the user has viewed or dismissed the notification.
 message UserNotificationState {
+   // kind is the resource kind ("user_notification_state").
+   string kind = 1;
+   // sub_kind is the optional resource subkind. This is unused.
+   string sub_kind = 2;
+   // version is the resource version.
+   string version = 3;
    // notification_state is the state of this notification for the user. This can be either "viewed" or "dismissed".
-   string notification_state = 1;
+   NotificationState notification_state = 4;
+}
+
+enum NotificationState {
+   // NOTIFICATION_STATE_VIEWED marks this notification has having been viewed (clicked on) by the user.
```

Comment:
Generally you want 0 to make sense as a default. I would do this:

```
enum NotificationState {
    NOTIFICATION_STATE_UNREAD = 0;
    NOTIFICATION_STATE_VIEWED = 1;
    NOTIFICATION_STATE_DISMISSED = 2;
```

Reply:
It should be noted that this value should never be set, we aren't keeping track of whether any individual notification is read or unread per-notification (we use `UserLastSeenNotification` for that), this is only for notifications that have been viewed (clicked on) or dismissed.

Reply:
:+1: I think "unread" still makes sense for enum value 0.